test/mechanics/timeIntegration/TimeControl.cpp

```diff
@@ -38,55 +38,119 @@ BOOST_AUTO_TEST_CASE(Debug_Fuctionality)
+    timeControl.SetMaxTimeStep(5.0);
+    expectedMsg = "Minimal timestep must be smaller than maximal Timestep!";
+    BOOST_CHECK_EXCEPTION(timeControl.SetMinTimeStep(6.0),
+                          MechanicsException,CheckException);
+    expectedMsg = "Maximal timestep must be bigger than minimal Timestep!";
+    BOOST_CHECK_EXCEPTION(timeControl.SetMaxTimeStep(0.5),
+                          MechanicsException,CheckException);
+
+
+    timeControl.SetTimeStepFunction([](TimeControl& timeControl,double,double,bool)->double{return timeControl.GetTimeStep();});
+    timeControl.SetTimeStep(1.0);
+    timeControl.Proceed();
+    timeControl.Proceed();
+
+    expectedMsg = "Final time must be larger than current time!";
+    BOOST_CHECK_EXCEPTION(timeControl.SetTimeFinal(1.0);,
+                          MechanicsException,CheckException);
+
+
+    expectedMsg = "No convergence with the current maximum number of "
+                  "iterations, either use automatic time stepping, "
+                  "reduce the time step or the minimal line search cut "
+                  "back factor. In case you provided a custom timestepping function "
+                  "and intend to use some kind of automatic timestepping, call "
+                  "the RestorePreviosTime() function of the time control before reducing "
+                  "the timestep.";
+    BOOST_CHECK_EXCEPTION(timeControl.AdjustTimestep(10,10,false);,
+                          MechanicsException,CheckException);
 }
 
 
 
 
 
 
-BOOST_AUTO_TEST_CASE(Equidistant_Timestepping)
+BOOST_AUTO_TEST_CASE(Timestepping)
 {
-    std::string expectedMsg = "";
-    std::function<bool(MechanicsException const&)> CheckException
-            = [&expectedMsg](MechanicsException const& ex)->bool
-                {
-                    return CheckExceptionMessageCorrect(ex,expectedMsg);
-                };
+    // Equidistant timestepping
     {
         TimeControl timeControl;
         timeControl.SetTimeFinal(20);
 
-        expectedMsg = "Timestep must be larger tham 0!";
-        BOOST_CHECK_EXCEPTION(timeControl.SetTimeStep(0),MechanicsException,CheckException);
-        BOOST_CHECK_EXCEPTION(timeControl.SetTimeStep(-1337.0),MechanicsException,CheckException);
 
-        timeControl.SetTimeStep(1);
-        timeControl.Proceed();
-        timeControl.Proceed();
+        timeControl.SetTimeStep(10.);
         timeControl.Proceed();
-        BOOST_CHECK_EQUAL(timeControl.GetCurrentTime(), 3.0);
+        BOOST_CHECK_EQUAL(timeControl.GetCurrentTime(), 10.0);
+        BOOST_CHECK_EQUAL(timeControl.Finished(), false);
 
-        timeControl.SetTimeStep(5.5);
+        timeControl.SetTimeStep(5.0);
+        timeControl.UseEquidistantTimestepping();
 
         timeControl.Proceed();
         timeControl.Proceed();
-        timeControl.Proceed();
-        BOOST_CHECK_EQUAL(timeControl.GetCurrentTime(), 19.5);
+        BOOST_CHECK_EQUAL(timeControl.GetCurrentTime(), 20.0);
```

Comment:
Dont know about all the boost test features yet. Will change it if  I don't forget to. :p